### PR TITLE
Remove Dependency on Xlib

### PR DESCRIPTION
**Pull request by shundhammer**
## Bugzilla

https://bugzilla.suse.com/show_bug.cgi?id=1182663

## Problem

The Qt UI plug-in cannot be loaded on aarch64.

## Cause

It has unresolved symbols from Xlib. On this architecture, libqt5 doesn't link against Xlib, that's why this is a problem on this architecture, but not on others.

## Fix

Removed all Xlib calls in libyui-qt. Most of them are leftovers from earlier times (back in the early 2000s) that are no longer necessary.

## Disabled: CapsLock Detection

We used to detect if CapsLock is on for password fields, replacing the password field caption with "CapsLock!" because the characters are not echoed as usual, so the user cannot see that he is only typing uppercase letters.

But that silently stopped working many years ago when Qt5 removed QWidget::x11Event(). We are now no longer reimplementing a virtual function that is called in the input event handlers, we are just implementing a new function that is never called anywhere.

We will need a replacement for this functionality, even though nobody noticed that it was missing through all those years.

This is independent of the Xlib dependency, though: This function was only using the XEvent struct to check for the CapsLock modifier. Still, it included some low-level X11 headers.

Qt is using XCB now instead of Xlib, so using Xlib would be wrong anyway these days.

### CODE REVIEW

**Review by wfeldt — APPROVED** [review]
lgtm

File: src/YQInputField.cc [hunk 1]
```diff
@@ -253,8 +246,21 @@ void YQInputField::clearCapsLockWarning()
...
+    // FIXME: This used to check CapsLock, but it silently stopped working from
+    //        Qt 5.x on. Nobody ever realized this, but we want this feature
+    //        back some day soon.
+    //
+    // The reason why this never failed to compile was that this is overwriting
+    // a virtual function from the QWidget parent class, but the parent class
+    // simply doesn't have that function anymore, so this is just implementing
+    // a new function that is never called; so this failed silently.
+    //
+    // -- 2020-02-25 shundhammer
```
Comment (rwx788) [inline src/YQInputField.cc:261]: Hehe, it's 2021 already ;)

Comment (wfeldt) [reply src/YQInputField.cc:261]: uhm, I completely missed that one :-)

Comment (shundhammer) [reply src/YQInputField.cc:261]: I guess it will take me until at least September to get used to this new number.
Why does it have to change so often? ;-)

Comment (wfeldt) [reply src/YQInputField.cc:261]: As time passes faster subjectively as you get older, I guess once you hit December it's not much worth borthering at all. :-)

**Pull request merged**